### PR TITLE
villain: init at 2.1.0

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by brianmcgillion** [comment]
Result of `nixpkgs-review pr 330857` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>villain</li>
  </ul>
</details>

File: pkgs/by-name/vi/villain/package.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  src = fetchFromGitHub {
+    owner = "t3l3machus";
+    repo = "Villain";
+    rev = "v${version}";
+    hash = "sha256-8MOpbyw4HEJMcv84bNkNLBSZfEmIm3RDSUi0s62t9ko=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (gador) [inline pkgs/by-name/vi/villain/package.nix:23]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/vi/villain/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/villain}
+    rm README.md requirements.txt LICENSE.md
+    cp -a * $out/share/villain/
+    makeWrapper ${python3}/bin/python $out/bin/villain \
+      --add-flags "$out/share/villain/Villain.py" \
+      --prefix PYTHONPATH : ${python3Packages.makePythonPath propagatedBuildInputs}
```
Comment (gador) [inline pkgs/by-name/vi/villain/package.nix:38]:
```suggestion
      --prefix PYTHONPATH : ${python3Packages.makePythonPath dependencies}
```

File: pkgs/by-name/vi/villain/package.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "High level stage 0/1 C2 framework that can handle multiple TCP socket & HoaxShell-based reverse shells, enhance their functionality with additional features and share them among connected sibling servers";
```
Comment (gador) [inline pkgs/by-name/vi/villain/package.nix:43]:
```suggestion
    description = "High level stage 0/1 C2 framework that can handle multiple TCP socket & HoaxShell-based reverse shells";
```

**Review by gador — CHANGES_REQUESTED** [review]
just a few nitpicks, otherwise looks good!

**Comment by D3vil0p3r** [comment]
> just a few nitpicks, otherwise looks good!

Fixed. Please give a look now.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 330857` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>villain</li>
</ul>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 330857` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>villain</li>
</ul>
</details>

**Review by gador — APPROVED** [review]

**Pull request merged**